ipld/resolver/src/service.rs

```diff
@@ -672,46 +674,77 @@ pub fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBox)> {
 
 async fn download_blob(
     iroh: Iroh,
-    hash: Hash,
+    seq_hash: Hash,
     size: u64,
     node_addr: NodeAddr,
 ) -> anyhow::Result<()> {
+    // Download top-level blob
     // Use an explicit tag so we can keep track of it
-    // TODO: this needs to be tagged with a "user id"
-    let tag = iroh::blobs::Tag(format!("stored-seq-{hash}").into());
-    let res = iroh
-        .blobs()
+    let tag = iroh::blobs::Tag(format!("stored-seq-{seq_hash}").into());
+    iroh.blobs()
         .download_with_opts(
-            hash,
+            seq_hash,
             iroh::client::blobs::DownloadOptions {
                 format: iroh::blobs::BlobFormat::Raw,
-                nodes: vec![node_addr],
+                nodes: vec![node_addr.clone()],
                 tag: iroh::blobs::util::SetTagOption::Named(tag),
                 mode: iroh::client::blobs::DownloadMode::Queued,
             },
         )
         .await?
         .await?;
 
-    if res.local_size + res.downloaded_size != size {
-        return Err(anyhow!(
-            "downloaded blob size {} does not match expected size {}",
-            res.local_size + res.downloaded_size,
-            size
-        ));
+    // Read top-level blob as a hash sequence
+    let res = iroh
+        .blobs()
+        .read_to_bytes(seq_hash)
+        .await
+        .map_err(|e| anyhow!("failed to read hash sequence object: {} {}", seq_hash, e))?;
+    let hash_seq = HashSeq::try_from(res)
+        .map_err(|e| anyhow!("failed to parse hash sequence object: {} {}", seq_hash, e))?;
+
+    // Download blobs in the hash sequence
+    for (i, hash) in hash_seq.iter().enumerate() {
+        let tmp_id = Uuid::new_v4();
+        let tmp_tag = iroh::blobs::Tag(format!("temp-{hash}-{tmp_id}").into());
+        let res = iroh
+            .blobs()
+            .download_with_opts(
+                hash,
+                iroh::client::blobs::DownloadOptions {
+                    format: iroh::blobs::BlobFormat::Raw,
+                    nodes: vec![node_addr.clone()],
+                    tag: iroh::blobs::util::SetTagOption::Named(tmp_tag.clone()),
+                    mode: iroh::client::blobs::DownloadMode::Queued,
+                },
+            )
+            .await?
+            .await?;
+        iroh.tags().delete(tmp_tag).await.ok();
```

Comment:
delete tmp tag. this works assuming the top-level tag (already present at this point) protects these blobs from GC.

Reply:
these can already be deleted after the creation of the hashseq, not just after downloading